kubernetes/README.md

```diff
@@ -0,0 +1,165 @@
+| |   |   |/var/lib/teleport|       |   | Resources      |           |
+| |   |   +^----------------+       |   |                |           |     K8S
+| |   |    |                        |   |                |      +----v----------------+
+| +---+----+------------------------+   |                |      |                     |
+|     |    |                            |                |      |    +--------------+ |
+|     |    |                            |                |      |    |              | |
+|     |    |   operator                 |                |      |    | Reconciler   | |
+| +---+----+----------------------------+--------+       |      |    | Loop         | |
+| |   |    |                            |        |       |      |    |              | |
+| |  ++----+-+                    +-----v----+   |       |      |    |              | |
+| |  |tctl   <--------------------> teleport <---+-------+------+---->              | |
+| |  +-------+    Setup U&R       | operator |   |       |      |    |              | |
+| |             Create Identity   +----------+   |       |      |    |              | |
+| |                                              |       |      |    +--------------+ |
+| |                                              |       |      |                     |
+| +----------------------------------------------+       |      +---------------------+
+|                                                        |
+|                                                        |
+|                                                        |
++--------------------------------------------------------+
+```
+
+## Running
+
+### Requirementes
+
+#### K8S cluster
+You can use `minikube` if you don't have a cluster yet.
+
+#### Operator's docker image
+You can obtain the docker image by building it yourself.
+
+We're planning to upload the image to quay.io, but for now you have to build it yourself with the following command: `make docker-build-kubernetes`.
+
+This creates a `teleport-operator:latest` image that we'll be using later on to setup de Operator.
+
+#### HELM chart from Teleport with the operator
+We are re-using the Teleport Cluster Helm chart but modifying to start the operator using the image above.
+
+This change is not in master, so you'll need to checkout the HELM charts from this branch
+`marco/plugins-teleport-operator-charts` (https://github.com/gravitational/teleport/pull/12144)
+
+#### Other tools
+We also need the following tools: `helm`, `kubectl` and `docker`
+
+### Running the operator
+
+Start `minikube` with `minikube start`.
+
+In order to build the Operator's image, we must use the docker daemon from the `minikube`:
+`eval $(minikube docker-env --shell bash)`.
+
+Change to the Project's root directory (`teleport-plugins`) and build the docker image:
+`make docker-build-kubernetes`.
+
+Meanwhile (or right after) we need to install the CRDs into the cluster.
+
+You can do so by issuing `make kubernetes-install`.
```

Comment:
I have mixed feelings about this one

I can see three approaches here:
- as is: clone and then install with `kustomize`
- move the CRDs into the chart (it would land here https://github.com/gravitational/teleport/pull/12144)
- add the CRDs installation and upgrade to the operator itself

I think we both agree that the first one is the worst, so let's focus on the other two options

Adding the CRDs into the chart has the following advantages:
- it's the recommended tool for the job (as you linked)

However, I think the disadvantages outweigh the list above:
- we will need to create a PR in `teleport` every time the `types.pb` changes
- currently, there's no way to upgrade the CRDs specs

Embedding the CRDs files and using the `teleport-operator` to install them will give us better control over upgrades of the spec.

I can see us going with any of the 3 options for now, but I think we will end up with the 3rd one.
My concern with going with a full fledged implementation of option number 3 is that it will increase the diff size of the PR.
My suggestion is to keep it as it is now, but open up a PR right after this one is merged to implement option number 3

What do you think?

Reply:
Agreed, thank you 🙏

I created a version which relies on the operator to install the CRDs at boot
If you want to play with it, you just need the following:
- checkout `teleport` at `marco/plugins-teleport-operator-charts`
- `minikube start`
```
helm install teleport-cluster <teleport repo location>/examples/chart/teleport-cluster \
                    --create-namespace --namespace teleport-cluster \
                    --set clusterName=teleport-cluster.teleport-cluster.svc.cluster.local \
                    --set operator=true --set teleportVersionOverride=9.3.2
```